pgmpy/factors/continuous/ContinuousFactor.py

```diff
@@ -0,0 +1,393 @@
+from __future__ import division
+
+import six
+import numpy as np
+import scipy.integrate as integrate
+
+
+class ContinuousFactor(object):
+    """
+    Base class for factors representing various multivariate
+    representations.
+    """
+    def __init__(self, variables, pdf):
+        """
+        Parameters
+        ----------
+        variables: list or array-like
+            The variables for wich the distribution is defined.
+
+        pdf: function
+            The probability density function of the distribution.
+
+        Examples
+        --------
+        >>> import numpy as np
+        >>> from scipy.special import beta
+        # Two variable drichlet ditribution with alpha = (1,2)
+        >>> def drichlet_pdf(x, y):
+        ...     return (np.power(x, 1)*np.power(y, 2))/beta(x, y)
+        >>> from pgmpy.factors import ContinuousFactor
+        >>> drichlet_factor = ContinuousFactor(['x', 'y'], drichlet_pdf)
+        >>> drichlet_factor.scope()
+        ['x', 'y']
+        >>> drichlet_factor.assignemnt(5,6)
+        226800.0
+        """
+        if not isinstance(variables, (list, tuple, np.ndarray)):
+            raise TypeError("variables: Expected type list or array-like, "
+                             "got type {var_type}".format(var_type=type(variables)))
+
+        if len(set(variables)) != len(variables):
+            raise ValueError("Variable names cannot be same.")
+
+        self.variables = list(variables)
+        self.pdf = pdf
+
+    def scope(self):
+        """
+        Returns the scope of the factor.
+
+        Returns
+        -------
+        list: List of variable names in the scope of the factor.
+
+        Examples
+        --------
+        >>> from pgmpy.factors import ContinuousFactor
+        >>> from scipy.stats import multivariate_normal
+        >>> normal_pdf = lambda x: multivariate_normal(x, [0, 0], [[1, 0], [0, 1]])
+        >>> phi = ContinuousFactor(['x1', 'x2'], normal_pdf)
+        >>> phi.scope()
+        ['x1', 'x2']
+        """
+        return self.variables
+
+    def assignment(self, *args):
```

Comment:
If we are allowing kwargs in ```self.pdf``` then, the pdf obtained after operations like ```reduce```,  ```marginalize```, ```_operate``` should also allow kwargs parameters.

Take for example the ```reduced_pdf``` in ``reduce```, it checks if there are args or there are kwargs parameters and then if there are both. Then it maps the arguments accordingly.  I think this would increase the overhead and complication on the already slow implemented methods.